src/es-module-shims.js

```diff
@@ -260,7 +260,9 @@ function resolveDeps (load, seen) {
       }
       // dynamic import
       else {
-        resolvedSource += `${source.slice(lastIndex, dynamicImportIndex + 6)}Shim(${source.slice(start, end)}, ${load.r && urlJsString(load.r)}`;
+        const assertionStart = start + importName.length + 2;
```

Comment:
For dynamic import expressions, `importName` isn't always defined. Instead we should use the `se` offset here.

Reply:
Alternatively the `e: end` value.

Reply:
Thanks, using `e: end` is really a good idea. I made changes to the PR, with some renaming to make it less confusing between `e` and `se`.